### PR TITLE
Refactor/cleanup jaf

### LINKED ISSUE #45: Investigate and potentially refactor Header component to remove Router logic
As noted [here](https://github.com/espressoGoddess/beam/pull/42#discussion_r1163453603)

### LINKED ISSUE #56: Investigate & refactor api calls functions

### LINKED ISSUE #63: BUG: stale data once journal entry is updated
`App.js` uses the initial render fetch to store all of the `userLogs`. Once a user updates their journal, that data is never refreshed.

Thinking of two potential solutions:
- Re-fetch within `JournalEntry` component (this seems most straightforward)
- Figure out a way to tie the update into `App` state and set state based on journal entry update

### LINKED ISSUE #76: Update seed file with decent informan
instead of "kiss a stranger on the mouth" and "Blah blah blah blah"

### PR DESCRIPTION
**Pull request by jfogiato**
### Files Changed:
`Honestly too many I'm sorry I'm not typing them all out but the changes are pretty straight forward when you scroll through`

### What Has Changed: 
- Fixed the stale data bug
- Refactored `apiCalls` to utilize just one function
- Added a linter and ran/fixed errors
- Cleaned up seed data for more realistic presentation going forward
- Updated `favicon.ico` for the tab header

### Notes: (includes questions, ideas for the future, anything you want to call out)
I think most of the changes here will be pretty straight forward - lots of little things. Let me know if you have anything you want to chat through!

### Closes: (any issues that it closes)

Closes #76, Closes #63, Closes #56, Closes #45,

### CODE REVIEW

File: client/src/Components/JournalEntry/JournalEntry.js [hunk 1]
```diff
@@ -3,9 +3,9 @@ import { Link } from "react-router-dom";
 import { DateTime } from 'luxon';
 import PropTypes from 'prop-types';
 import './JournalEntry.css';
-import { updateEntry } from "../../utilities/api-calls";
+import { fetchCall } from "../../utilities/api-calls";
 
-const JournalEntry = ({ journal, entryID }) => {
+const JournalEntry = ({ journal, entryID, updateLogs }) => {
```
Comment (espressoGoddess) [inline client/src/Components/JournalEntry/JournalEntry.js:8]: hey Joe! Would you mind adding `updateLogs` to  the propTypes? 🫶🏼 It looks like `entryId` could be added as well, (Maybe i missed this one?)

File: client/public/index.html [hunk 2]
```diff
@@ -16,7 +16,7 @@
     />
     <link rel="apple-touch-icon" href="%PUBLIC_URL%/logo192.png" />
     <link rel="manifest" href="%PUBLIC_URL%/manifest.json" />
-    <title>React App</title>
+    <title>Beam</title>
```
Comment (espressoGoddess) [inline client/public/index.html:19]: yeeeehooo! nice, thanks

File: client/src/Components/Activities/Activities.js [hunk 3]
```diff
@@ -9,7 +9,7 @@ const Activities = ({ activities, addEntry, setFeeling }) => {
   const { params } = useRouteMatch('/what-should-you-do/:feeling');
   useEffect(() => {
     setFeeling(params.feeling);
-  }, [setFeeling, params.feeling])
+  }, [setFeeling, params.feeling]);
```
Comment (espressoGoddess) [inline client/src/Components/Activities/Activities.js:12]: nice catch 👍🏽

File: server/data/entryData.js [hunk 4]
```diff
@@ -10,64 +10,64 @@ const entryData = [
...
+    "activity": "Set a 5, 10, or 15-minute timer and tidy your space in the easiest way possible. You can stop when it goes off even mid-task. Anything counts. Nice work!",
+    "journal_entry": "I just can't seem to get it together. Will I ever figure it out???"
   },
   {
     "user_id": 1,
     "date": "2023-04-03",
     "feeling": "frustrated",
     "activity": "Go find a spider.",    
-    "journal_entry": "Today I am feeling anxious because I pooped my pants in front of my crush."
+    "journal_entry": "I got a flat tire on the way to work and then ripped my pants when I was trying to change it."
```
Comment (espressoGoddess) [inline server/data/entryData.js:35]: 🙀

**Review by espressoGoddess — COMMENTED** [review]
Great attention to detail Joe!

File: server/data/entryData.js [hunk 5]
```diff
@@ -10,64 +10,64 @@ const entryData = [
...
   },
   {
     "user_id": 1,
     "date": "2023-04-10",
-    "feeling": "happy",
-    "activity": "Be more sad.",
-    "journal_entry": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."
+    "feeling": "bad",
+    "activity": "Create a whimsical talisman: Craft a silly and unusual object or trinket that represents your bad mood. This can be anything from a miniature rubber chicken to a glittery rock. Keep it with you as a reminder to take things less seriously and embrace the playful side of life.",
+    "journal_entry": "I wasn't feeling great, but then I created a whimsical talisman and felt much better! I edited this entry after :) Ever seen Hereditary? My talisman was similar to the dolls that 'little girl' made."
```
Comment (chrissycooper) [inline server/data/entryData.js:70]: 😨 I HAVE seen Hereditary and now I'm scared of our user........  Anyway, I love these updates, Joe. Such curious and impactful storytelling in just a few sentences.

File: client/src/sampleData/activityData.js [hunk 6]
```diff
@@ -1,209 +1,213 @@
...
+    "activity": 'Set a 5, 10, or 15-minute timer and tidy your space in the easiest way possible. You can stop when it goes off even mid-task. Anything counts. Nice work!',
+    "feelings": [ 'bad' ]
   },
   {
-    activityId: 41,
-    activity: 'Put on an irresistible song and wiggle your body the weirdest way you can. Decide on the movements based on listening to how your body wants to move, stretch, and shake out, NOT on if the dance moves are good. The more like Ace Ventura: Pet Detective meets hippie at Woodstock the better',
-    feelings: [ 'bad' ]
+    "activityId": 41,
+    "activity": 'Put on an irresistible song and wiggle your body the weirdest way you can. Decide on the movements based on listening to how your body wants to move, stretch, and shake out, NOT on if the dance moves are good. The more like Ace Ventura: Pet Detective meets hippie at Woodstock the better',
+    "feelings": [ 'bad' ]
```
Comment (chrissycooper) [inline client/src/sampleData/activityData.js:210]: WHEW thanks for all of those

**Review by chrissycooper — APPROVED** [review]

File: client/src/Components/App/App.js [hunk 7]
```diff
@@ -49,10 +49,14 @@ const App = () => {
   }, [feeling]);
 
   useEffect(() => {
-    getAllEntries()
-    .then(data => setUserLogs(data));
+    updateLogs();
   }, []);
 
+  const updateLogs = () => {
```
Comment (chrissycooper) [inline client/src/Components/App/App.js:55]: Love this function, very semantic/clear what it's purpose is. Thank you for keeping our data model in tact.

File: client/public/index.html (hunk 2 above)
Comment (KatherineBlaine) [reply client/public/index.html:19]: Good catch

File: client/src/Components/Feelings/Feelings.js [hunk 8]
```diff
@@ -9,12 +9,12 @@ const Feelings = ({ setFeeling }) => {
...
     return feelingData.associatedFeelings.map(feel => {
       return <Link to={`/why-are-you-feeling-that-way`} key={feel} className='uni-btn' onClick={()=>setFeeling(feel)}>{feel}</Link>
-    })
+    });
   });
 
   const sortedLinks = feelingsLinks.flat().sort((a, b) => {
     return (a.key > b.key) ? 1 : -1;
-  })
+  });
```
Comment (KatherineBlaine) [inline client/src/Components/Feelings/Feelings.js:17]: Semicolon king 👑

File: client/src/utilities/api-calls.js [hunk 9]
```diff
@@ -1,40 +1,15 @@
...
-      return response.json();
-    } else {
-      throw new Error(`HTTP Status Code: ${response.status}`);
-    }
-  })
-  .catch(error => console.log(error));
-}
+const fetchCall = (entry, id) => {
+  const method = entry && id ? "PUT" : entry ? "POST" : "GET";
+  const body = entry?.date ? entry : entry ? {journal_entry: entry} : null;
```
[the remark this answers is not in the document]
Comment (KatherineBlaine) [reply client/src/utilities/api-calls.js:3]: Hmm I hear you Chrissy. I am super impressed with the clean logic here though, Joe. I could go either way!

File: server/data/entryData.js [hunk 10]
```diff
@@ -10,64 +10,64 @@ const entryData = [
...
+    "journal_entry": "I got a flat tire on the way to work and then ripped my pants when I was trying to change it."
   },
   {
     "user_id": 1,
     "date": "2023-04-02",
     "feeling": "nervous",
-    "activity": "Be cool.",
-    "journal_entry": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."
+    "activity": "Punch and hit your pillow as much as you want to work out some energy.",
+    "journal_entry": "Recent weather events have me feeling pretty nervous about the future of our planet. I wonder if policy makers will get their shit together?"
```
Comment (KatherineBlaine) [inline server/data/entryData.js:42]: Such an insightful journal entry

**Review by KatherineBlaine — APPROVED** [review]
Thanks for this cleanup, Joe! Looks amazing.

File: client/src/utilities/api-calls.js (hunk 9 above)
[the remark this answers is not in the document]
Comment (jfogiato) [reply client/src/utilities/api-calls.js:3]: @chrissycooper Hahaha I think that's a totally fair assessment!! I think my two main reasons for this were (1) it was honestly a really fun challenge lol, and (2) I've gotten feedback on instances really similar to this that I needed to DRY up code. That being said, there may be a way to get closer to riding the line between the two - DRYing && legibility. 

Super appreciate your feedback on this 🧡

File: client/src/Components/JournalEntry/JournalEntry.js (hunk 1 above)
Comment (jfogiato) [reply client/src/Components/JournalEntry/JournalEntry.js:8]: @espressoGoddess For sure!! I'm not as familiar with PropTypes so would love to chat on this quickly before I push up that change

**Pull request merged**